### ISSUE TITLE
Investigate if kubernetes descheduler is something we want

### ISSUE THREAD

**Issue by Xartos** [elastisys/compliantkubernetes-apps#1336]
**What should be investigated.**
What technology/product/concept should be investigated?

https://github.com/kubernetes-sigs/descheduler

Now when we have kured rebooting nodes we might get into situations where customers get unbalanced workload on their nodes.

It would be cool to check if we could use something like this to be able to rebalance the cluster and hopefully make it run smoother.

**What artifacts should this produce.**
How should the knowledge gathered from this task be captured.

* Investigation doc describing the complexity of setting up and operating.

**Comment by crssnd** [comment]
@cristiklein can you please attach the thesis results for this topic

**Comment by cristiklein** [comment]
Please find attached Oliver Larsson's thesis on descheduler.

I'm copy-pasting the abstract for convenience:

> Over time, as applications scale up and down in a Kubernetes-managed environment, a cluster state of resource 
 fragmentation may be reached. Such a state indicates that the hardware of a cluster is unable to be utilized to its fullest potential, as resources become locked by poor Pod to Node mappings. This thesis shows that such problems are not uncommon as workload requirements approach the resource availability of a cluster. Additionally, an experimental analysis of directed Pod eviction is carried out. Directed Pod eviction is a technique that attempts to combat the issue of resource 
 fragmentation. It is found through this analysis that directed Pod eviction can reduce resource fragmentation and allow clusters to run the same workload using 16.7% fewer Nodes than what is consistently possible using standard Kubernetes scheduling.

TL;DR: Descheduler does not help if the cluster has insufficient capacity and it does not help if the cluster has plenty of capacity. However, it might save a Node if the workload is close to consuming the cluster's capacity.

[thesis.pdf](https://github.com/elastisys/compliantkubernetes-apps/files/10410720/thesis.pdf)
